packages/client/src/generation/TSClient/Args.ts

```diff
@@ -1,144 +1,118 @@
-import indent from 'indent-string'
-
 import { DMMF } from '../dmmf-types'
-import { getIncludeName, getLegacyModelArgName, getModelArgName, getSelectName } from '../utils'
-import { TAB_SIZE } from './constants'
-import type { Generatable } from './Generatable'
+import * as ts from '../ts-builders'
+import {
+  extArgsParam,
+  getExcludeName,
+  getIncludeName,
+  getLegacyModelArgName,
+  getModelArgName,
+  getSelectName,
+} from '../utils'
 import { GenerateContext } from './GenerateContext'
 import { getArgFieldJSDoc } from './helpers'
-import { InputField } from './Input'
+import { buildInputField } from './Input'
+
+export class ArgsTypeBuilder {
```

Comment:
This replaces both `ArgsType` and `MinArgsType` type classes - we now can selectively decide how to build different args types and which fields to include.